src/averageAges.js

```diff
@@ -55,7 +64,17 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(person => !onlyWithSon
+    ? people.find(child => child.name === person.mother)
+    : people.some(child => child.name === person.mother) && person.sex === 'm'
+  );
+
+  const diff = children.map(kid =>
```

Comment:
```suggestion
  const ageGaps = children.map(kid =>
```
название для массива давай во множественном числе